### ISSUE TITLE
randomFloat edge values problem

### ISSUE THREAD

**Issue by sgrishanin** [fzaninotto/Faker#1856]
### Summary

For my unit tests I generate decimal numbers up to 99.999999 with variable number of decimals. Sometimes it returns 100.0. See the example below.
Apparently it first generates the number up to the max, and then rounds it to the given nbMaxDecimals instead of straight up cutting. I've searched through the release notes for 1.9 and it doesn't look like this issue has been addressed.
### Versions
|                     | Version |
|:--------------------|:--------|
| PHP                 | 7.3.11   |
| `fzaninotto/faker`  | 1.8.0   |

### Self-enclosed code snippet for reproduction

To easily reproduce the bug min is equal to max
```php
$decimals = 1;
$min = 99.999999;
$max = 99.999999;
$oneHundred = $faker->randomFloat($decimals, $min, $max);
```

### Expected output

```txt
99.9
```

### Actual output

```txt
100.0
```

**Comment by pimjansen** [comment]
I dont see this is a bug? You asked it to give you 1 decimal so it has to round the value. In this case .99 will be 0

**Comment by sgrishanin** [comment]
@pimjansen Sure, it does round, but I don't want it to. I understand this is kind of arguable, but I think if this method takes $max as an argument, I expect it to give a number less or equal to that variable. Saying "it has to round the value" implies that I should be aware of its internals. 
I've mentioned my use case, how do you think I should go around this?

**Comment by pimjansen** [comment]
@sgrishanin well its a breaking change and the current behavior is exactly as expected. @localheinz what do you think about this?

I agree that the value should never be greater than the `$max` though

**Comment by sgrishanin** [comment]
@pimjansen yeah, I'm not saying it's a bug per se, that's just a little inconsistent to me.
